substrate/runtime/contract/src/vm.rs

```diff
@@ -0,0 +1,785 @@
+	) -> Result<sandbox::ReturnValue, sandbox::HostError> {
+		let transfer_to_ptr = args[0].as_i32().unwrap() as u32;
+		let transfer_to_len = args[1].as_i32().unwrap() as u32;
+		let value_ptr = args[2].as_i32().unwrap() as u32;
+		let value_len = args[3].as_i32().unwrap() as u32;
+
+		let mut transfer_to = Vec::new();
+		transfer_to.resize(transfer_to_len as usize, 0);
+		e.memory().get(transfer_to_ptr, &mut transfer_to)?;
+		let transfer_to = T::AccountId::decode(&mut &transfer_to[..]).unwrap();
+
+		let mut value_buf = Vec::new();
+		value_buf.resize(value_len as usize, 0);
+		e.memory().get(value_ptr, &mut value_buf)?;
+		let value = T::Balance::decode(&mut &value_buf[..]).unwrap();
+
+		// TODO: Read input data from memory.
+		let input_data = Vec::new();
+
+		// TODO: Let user to choose how much gas to allocate for the execution.
+		let nested_gas_limit = e.gas_meter.gas_left();
+		let ext = &mut e.ext;
+		let call_outcome = e.gas_meter.with_nested(nested_gas_limit, |nested_meter| {
+			match nested_meter {
+				Some(nested_meter) => ext.call(&transfer_to, value, nested_meter, input_data),
+				None => {
+					// there is not enough gas to allocate for the nested call.
+					Err(())
+				}
+			}
+		});
+
+		match call_outcome {
+			Ok(CallReceipt { .. }) => {
+				// TODO: Find a way how to pass return_data back to the this sandbox.
+				Ok(sandbox::ReturnValue::Unit)
+			}
+			Err(_) => {
+				// TODO: Return a status code value that can be handled by the caller instead of a trap.
+				Err(sandbox::HostError)
+			}
+		}
+	}
+
+	// ext_create(code_ptr: u32, code_len: u32, value_ptr: u32, value_len: u32)
+	fn ext_create<T: Trait, E: Ext<T>>(
+		e: &mut Runtime<T, E>,
+		args: &[sandbox::TypedValue],
+	) -> Result<sandbox::ReturnValue, sandbox::HostError> {
+		let code_ptr = args[0].as_i32().unwrap() as u32;
+		let code_len = args[1].as_i32().unwrap() as u32;
+		let value_ptr = args[2].as_i32().unwrap() as u32;
+		let value_len = args[3].as_i32().unwrap() as u32;
+
+		let mut value_buf = Vec::new();
+		value_buf.resize(value_len as usize, 0);
+		e.memory().get(value_ptr, &mut value_buf)?;
+		let value = T::Balance::decode(&mut &value_buf[..]).unwrap();
+
+		let mut code = Vec::new();
+		code.resize(code_len as usize, 0u8);
+		e.memory().get(code_ptr, &mut code)?;
+
+		// TODO: Read input data from the sandbox.
+		let input_data = Vec::new();
+
+		// TODO: Let user to choose how much gas to allocate for the execution.
+		let nested_gas_limit = e.gas_meter.gas_left();
+		let ext = &mut e.ext;
+		let create_outcome = e.gas_meter.with_nested(nested_gas_limit, |nested_meter| {
+			match nested_meter {
+				Some(nested_meter) => ext.create(&code, value, nested_meter, input_data),
+				None => {
+					// there is not enough gas to allocate for the nested call.
+					Err(())
+				}
+			}
+		});
+
+		match create_outcome {
+			Ok(CreateReceipt { .. }) => {
+				// TODO: Copy an address of the created contract in the sandbox.
```

Comment:
same as above.

Reply:
let's move this discussion in [this thread](https://github.com/paritytech/polkadot/pull/345#discussion_r203060428).